src/systemd/_sd-common.h

```diff
@@ -75,4 +75,11 @@
 #  endif
 #endif
 
+#define _SD_DEFINE_POINTER_CLEANUP_FUNC(type, func)             \
+        static inline void func##p(type **p) {                  \
+                if (*p)                                         \
+                        func(*p);                               \
+        }                                                       \
+        struct __useless_struct_to_allow_trailing_semicolon__
```

Comment:
(in particular as this is not the only place we use such a struct in public APIs)